src/draw/sw/lv_draw_sw_rect.c

```diff
@@ -96,6 +117,159 @@ void lv_draw_sw_rect(const lv_area_t * coords, const lv_area_t * clip, const lv_
 /**********************
  *   STATIC FUNCTIONS
  **********************/
+#if DITHER_GRADIENT
+LV_ATTRIBUTE_FAST_MEM static inline uint8_t clamp(int a)
+{
+    return (uint8_t)(a < 0 ? 0 : (a > 255 ? 255 : a));
```

Comment:
We already have `LV_CLAMP(min, value, max)`.

Reply:
Right, will use that